sulong/projects/com.oracle.truffle.llvm.runtime/src/com/oracle/truffle/llvm/runtime/pointer/CommonPointerLibraries.java

```diff
@@ -116,6 +142,84 @@ static boolean isMemberModifiable(LLVMPointerImpl receiver, String ident,
         }
     }
 
+    @ExportMessage
+    static boolean isMemberInvocable(LLVMPointerImpl receiver, String ident) {
+        LLVMInteropType type = receiver.getExportType();
+        if (type instanceof LLVMInteropType.Clazz) {
+            LLVMInteropType.Clazz clazz = (LLVMInteropType.Clazz) type;
+            return clazz.findMethod(ident) != null;
+        }
+        return false;
+    }
+
+    @ExportMessage
+    static class InvokeMember {
+        @SuppressWarnings("unused")
+        @Specialization(guards = {"asClazz(receiver)==clazz", "member.equals(methodName)", "argCount==arguments.length"})
+        static Object doCached(LLVMPointerImpl receiver, String member, Object[] arguments,
+                        @CachedContext(LLVMLanguage.class) LLVMContext context, @CachedLibrary(limit = "5") InteropLibrary interop,
+                        @Cached(value = "asClazz(receiver)", allowUncached = true) LLVMInteropType.Clazz clazz,
+                        @Cached(value = "clazz.findMethodByArguments(receiver, member, arguments)", allowUncached = true) Method method,
+                        @Cached(value = "arguments.length", allowUncached = true) int argCount,
+                        @Cached(value = "method.getName()", allowUncached = true) String methodName,
+                        @Cached(value = "getLLVMFunction(context, method, clazz, member)", allowUncached = true) LLVMFunction llvmFunction,
+                        @Cached(value = "create(llvmFunction)", allowUncached = true) LLVMAccessSymbolNode accessSymbolNode)
+                        throws UnsupportedMessageException, ArityException, UnsupportedTypeException {
+            Object[] newArguments = addSelfObject(receiver, arguments);
+            return interop.execute(accessSymbolNode.execute(), newArguments);
+        }
+
+        @SuppressWarnings("unused")
+        @Specialization
+        static Object doResolve(LLVMPointerImpl receiver, String member, Object[] arguments,
+                        @CachedContext(LLVMLanguage.class) LLVMContext context, @CachedLibrary(limit = "5") InteropLibrary interop,
+                        @Cached(value = "asClazz(receiver)", allowUncached = true) LLVMInteropType.Clazz clazz,
+                        @Cached(value = "clazz.findMethodByArguments(receiver, member, arguments)", allowUncached = true) Method method,
+                        @Cached(value = "arguments.length", allowUncached = true) int argCount,
+                        @Cached(value = "method.getName()", allowUncached = true) String methodName,
+                        @Cached(value = "getLLVMFunction(context, method, clazz, member)", allowUncached = true) LLVMFunction llvmFunction,
+                        @Cached(value = "create(llvmFunction)", allowUncached = true) LLVMAccessSymbolNode accessSymbolNode)
```

Comment:
These `@Cached` values are all unused. I don't think they are necessary, since this is the generic fallback specialization that does the resolving on each call.